app/routes/goal_routes.py

```diff
@@ -1 +1,83 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from .route_utilities import create_model, get_models_with_filters, validate_model
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    return create_model(Goal, request_body)
+
+@bp.get("")
+def get_all_goals():
+    return get_models_with_filters(Goal, request.args)
+
+@bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    result = {}
+    result["goal"] = goal.to_dict()
+    return result
+
+@bp.put("/<goal_id>")
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+    
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
+
+@bp.delete("/<goal_id>")
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    db.session.delete(goal)
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
+
+@bp.post("/<goal_id>/tasks")
+def connect_task_to_goal(goal_id):
+    
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json() 
+    tasks = []
+
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        tasks.append(task)
+    # If any of the task IDs are invalid, will abort before this point.
+    # Assume we do not want to associate any of the tasks with the goal if *any*
+    # of the IDs are invalid.
+    
+    for task in goal.tasks:
+        # Unassociate previously associated tasks
+        task.goal_id = None
+
+    for task in tasks:
+        task.goal_id = goal_id # only need to update FK in task to create relationship
+
+    db.session.commit()
```

Comment:
@ellenjin, remember that we used [relationship function](https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship) on our models. This allows us to more easily to navigate the relationship between two models by using their respective relationship attributes. Instead of looping through the tasks individually we could simply leverage the `.tasks` attribute, a list of associated `Tasks` like so: 
```py
    valid_tasks = []

    for task in task_ids_list:
        task = validate_model(Task, task)
        valid_tasks.append(task) 
    
    goal.tasks = valid_tasks
    db.session.commit()
```